Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -0,0 +1,86 @@
+/// <reference path="../../../definitions/vsts-task-lib.d.ts" />
+
+import tl = require('vsts-task-lib/task');
+import {ToolRunner} from 'vsts-task-lib/toolrunner';
+
+// Creates the tool runner for executing SonarQube.
+export function applySonarQubeParams(toolRunner:ToolRunner, sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?, sqDbUsername?, sqDbPassword?):ToolRunner {
+
+    toolRunner.arg('-Dsonar.host.url=' + sqHostUrl);
+    toolRunner.arg('-Dsonar.login=' + sqHostUsername);
+    toolRunner.arg('-Dsonar.password=' + sqHostPassword);
+    if (sqDbUrl) {
+        toolRunner.arg('-Dsonar.jdbc.url=' + sqDbUrl);
+    }
+    if (sqDbUsername) {
+        toolRunner.arg('-Dsonar.jdbc.username=' + sqDbUsername);
+    }
+    if (sqDbPassword) {
+        toolRunner.arg('-Dsonar.jdbc.password=' + sqDbPassword);
+    }
+
+    return toolRunner;
+}
+
+export class SonarQubeEndpoint {
+    constructor(public Url, public Username, public Password) {
+    }
+}
+
+// Gets SonarQube connection endpoint details.
+export function getSonarQubeEndpointDetails(inputFieldName):SonarQubeEndpoint {
+    var errorMessage = "Could not decode the generic endpoint. Please ensure you are running the latest agent (min version 0.3.2)";
+    if (!tl.getEndpointUrl) {
+        throw new Error(errorMessage);
+    }
+
+    var genericEndpoint = tl.getInput(inputFieldName);
+    if (!genericEndpoint) {
+        throw new Error(errorMessage);
+    }
+
+    var hostUrl = tl.getEndpointUrl(genericEndpoint, false);
+    if (!hostUrl) {
+        throw new Error(errorMessage);
+    }
+
+    // Currently the username and the password are required, but in the future they will not be mandatory
+    // - so not validating the values here
+    var hostUsername = getSonarQubeAuthParameter(genericEndpoint, 'username');
+    var hostPassword = getSonarQubeAuthParameter(genericEndpoint, 'password');
+    tl.debug("hostUsername: " + hostUsername);
```

Comment:
You can delete this line, there's no point in outputting this if it will be masked.